src/nvim/eval/funcs.c

```diff
@@ -6022,17 +6022,17 @@ static void f_readfile(typval_T *argvars, typval_T *rettv, EvalFuncData fptr)
         // small, to avoid repeatedly 'allocing' large and
         // 'reallocing' small.
         if (prevsize == 0) {
-          prevsize = (long)(p - start);
+          prevsize = (long)(p - start);  // NOLINT(google-readability-casting)
         } else {
           long grow50pc = (prevsize * 3) / 2;
-          long growmin  = (long)((p - start) * 2 + prevlen);
+          long growmin  = (long)((p - start) * 2 + prevlen);  // NOLINT(google-readability-casting)
           prevsize = grow50pc > growmin ? grow50pc : growmin;
         }
         prev = xrealloc(prev, (size_t)prevsize);
       }
       // Add the line part to end of "prev".
       memmove(prev + prevlen, start, (size_t)(p - start));
-      prevlen += (long)(p - start);
+      prevlen += (long)(p - start);  // NOLINT(google-readability-casting)
```

Comment:
It's also possible to change `prevlen` `prevsize` `grow50pc` `growmin` all to `ptrdiff_t` and remove this cast.